src/core/context/AppContextProvider.tsx

```diff
@@ -543,12 +546,39 @@ async function syncWithLocalStorage(
   );
 
   const existingConversation = allConversations[conversationID];
+
+  let tokensRemaining = contextLength;
+
+  const isDeepseekModel = id.includes('deepseek');
+  const isGPTModel = id.includes('gpt');
+
+  //  todo - refactor to use modelConfig for both and simplify conditionals
+  if (isDeepseekModel) {
+    //  If we get a response with usage, use that to calculate
+  } else {
+    //  todo - plug in deepseek approximation function
+  }
```

Comment:
This PR shows a preview of how this will work: https://github.com/Kava-Labs/oros/blob/92220c8ad70acba29338b2ba6bac89234489e30b/src/core/context/AppContextProvider.tsx#L435